### PR TITLE
🌱ROSA Cleanup & Documentation

### PR DESCRIPTION
**Pull request by muraee**
**What type of PR is this?**

/kind cleanup
/kind documentation

**What this PR does / why we need it**:
follow-up of https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4453

This PR:
- Introduce missing steps to deploy ROSA CRDs and enable ROSA controller when installing AWS provider.
- Add documentation on how to enable ROSA support and create ROSA clusters using the AWS provider.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes [#4428](https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/4428)

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits
- [x] includes documentation
- [ ] includes [emojis](https://github.com/kubernetes-sigs/kubebuilder-release-tools?tab=readme-ov-file#kubebuilder-project-versioning)
- [ ] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: controlplane/rosa/controllers/rosacontrolplane_controller.go [hunk 1]
```diff
@@ -291,6 +292,10 @@ func (r *ROSAControlPlaneReconciler) reconcileNormal(ctx context.Context, rosaSc
 
 	// Create the connection, and remember to close it:
 	token := os.Getenv("OCM_TOKEN")
+	ocmAPIUrl := os.Getenv("OCM_API_URL")
+	if ocmAPIUrl == "" {
+		ocmAPIUrl = "https://api.stage.openshift.com"
```
Comment (enxebre) [inline controlplane/rosa/controllers/rosacontrolplane_controller.go:297]: let's default to prod

File: controlplane/rosa/controllers/rosacontrolplane_controller.go [hunk 2]
```diff
@@ -338,6 +343,10 @@ func (r *ROSAControlPlaneReconciler) reconcileDelete(_ context.Context, rosaScop
 	// Create the connection, and remember to close it:
 	// TODO: token should be read from a secret: https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/4460
 	token := os.Getenv("OCM_TOKEN")
+	ocmAPIUrl := os.Getenv("OCM_API_URL")
```
Comment (enxebre) [inline controlplane/rosa/controllers/rosacontrolplane_controller.go:346]: let's default to prod

Comment (muraee) [reply controlplane/rosa/controllers/rosacontrolplane_controller.go:346]: fixed

File: templates/cluster-template-rosa.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+metadata:
+  name: "${CLUSTER_NAME}-control-plane"
+spec:
+  version: "${OPENSHIFT_VERSION}"
+  region: "${AWS_REGION}"
+  accountID: "${AWS_ACCOUNT_ID}"
+  creatorARN: "${AWS_CREATOR_ARN}"
+  machineCIDR: "10.0.0.0/16"
+  rolesRef:
+    ingressARN: "arn:aws:iam::${AWS_ACCOUNT_ID}:role/${OPERATER_ROLES_PREFIX}-ingress-operator-cloud-credentials"
```
Comment (enxebre) [inline templates/cluster-template-rosa.yaml:36]: ingress, registry, storageand. network should be `${OPERATER_ROLES_PREFIX}-openshift-...`

File: templates/cluster-template-rosa.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    nodePoolManagementARN: "arn:aws:iam::${AWS_ACCOUNT_ID}:role/${OPERATER_ROLES_PREFIX}-kube-system-capa-controller-manager"
+    controlPlaneOperatorARN: "arn:aws:iam::${AWS_ACCOUNT_ID}:role/${OPERATER_ROLES_PREFIX}-kube-system-control-plane-operator"
+    kmsProviderARN: "arn:aws:iam::${AWS_ACCOUNT_ID}:role/${OPERATER_ROLES_PREFIX}-kube-system-kms-provider"
+  oidcID: "${OIDC_CONFIG_ID}"
+  subnets:
+    - "${PUBLIC_SUBNET_ID}"
+    - "${PRIVATE_SUBNET_ID}"
+  availabilityZones:
+    - "${AWS_AVAILABILITY_ZONE}"
+  installerRoleARN: "arn:aws:iam::${AWS_ACCOUNT_ID}:role/${ACCOUNT_ROLES_PREFIX}-Installer-Role"
```
Comment (enxebre) [inline templates/cluster-template-rosa.yaml:50]: installer and support should be `${ACCOUNT_ROLES_PREFIX}-HCP-ROSA-...`

File: docs/book/src/topics/rosa/index.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+## Overview
+
+The AWS provider supports creating Red Hat OpenShift Service on AWS ([ROSA](https://www.redhat.com/en/technologies/cloud-computing/openshift/aws)) based cluster. Currently the following features are supported:
+
+- Provisioning/Deleting a ROSA cluster with hosted control planes ([HCP](https://docs.openshift.com/rosa/rosa_hcp/rosa-hcp-sts-creating-a-cluster-quickly.html))
+
+The implementation introduces the following CRD kinds:
+
+- ROSAControlPlane - specifies the ROSA Cluster in AWS
+- ROSACluster - needed only to statisfy cluster-api contract
```
Comment (AndiDog) [inline docs/book/src/topics/rosa/index.md:15]:
```suggestion
- `ROSAControlPlane` - specifies the ROSA Cluster in AWS
- `ROSACluster` - needed only to satisfy cluster-api contract
```

**Review by AndiDog — COMMENTED** [review]
LGTM except for one minor typo

File: docs/book/src/topics/rosa/index.md (hunk 5 above)
Comment (muraee) [reply docs/book/src/topics/rosa/index.md]: thanks, I fixed that.

**Review by AndiDog — COMMENTED** [review]
/lgtm

**Comment by muraee** [comment]
@AndiDog sorry I forgot to add a small change, could you retag it again please

**Review by AndiDog — COMMENTED** [review]
/lgtm

but to fix the tests, we first need https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4710 done.

**Review by vincepri — APPROVED** [review]
/approve
/lgtm

**Pull request merged**